File: problems/easy/programmer_day/__tests__/TestProgrammerDay.java
package problems.easy.programmer_day.__tests__;

import problems.easy.programmer_day.ProgrammerDay;

public class TestProgrammerDay {
    public static void main(String[] args) {
        int year = 1984;
        String result = ProgrammerDay.dayOfProgrammer(year);

        if (!result.equals("12.09.1984")) {
            throw new AssertionError("❌ Test failed. Expected 12.09.1984, got " + result);
        }

        System.out.println("✅ Java test passed!");
    }
}

File: problems/easy/programmer_day/__tests__/test_programmer_day.py
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from day_of_programmer import dayOfProgrammer

def test_day_of_programmer():
    year = 1984
    result = dayOfProgrammer(year)
    assert result == "12.09.1984", f"Expected 12.09.1984, got {result}"

    print("✅ Python test passed!")

if __name__ == "__main__":
    test_day_of_programmer()
